pkg/storage/stores/indexshipper/compactor/deletion/delete_requests_manager.go

```diff
@@ -199,52 +227,47 @@ func (d *DeleteRequestsManager) MarkPhaseFailed() {
 	d.deleteRequestsToProcessMtx.Lock()
 	defer d.deleteRequestsToProcessMtx.Unlock()
 
-	d.deleteRequestsToProcess = d.deleteRequestsToProcess[:0]
+	d.deleteRequestsToProcess = map[string]*userDeleteRequests{}
 }
 
 func (d *DeleteRequestsManager) MarkPhaseFinished() {
 	d.deleteRequestsToProcessMtx.Lock()
 	defer d.deleteRequestsToProcessMtx.Unlock()
 
-	for _, deleteRequest := range d.deleteRequestsToProcess {
-		if err := d.deleteRequestsStore.UpdateStatus(context.Background(), deleteRequest.UserID, deleteRequest.RequestID, StatusProcessed); err != nil {
-			level.Error(util_log.Logger).Log(
-				"msg", "failed to mark delete request for user as processed",
-				"delete_request_id", deleteRequest.RequestID,
-				"user", deleteRequest.UserID,
-				"err", err,
-				"deleted_lines", deleteRequest.DeletedLines,
-			)
-		} else {
-			level.Info(util_log.Logger).Log(
-				"msg", "delete request for user marked as processed",
-				"delete_request_id", deleteRequest.RequestID,
-				"user", deleteRequest.UserID,
-				"deleted_lines", deleteRequest.DeletedLines,
-			)
+	for _, userDeleteRequests := range d.deleteRequestsToProcess {
+		for _, deleteRequest := range userDeleteRequests.requests {
```

Comment:
I think `userDeleteRequests` could be nil here; we should add a guard to prevent a panic

Reply:
we always add a non-nil entry, did I miss anything?

Reply:
Sure, but later when we refactor this code and someone sets a nil value on purpose or by accident, this will panic. Better to be defensive IMHO